OSOE-93: Add a UI test to Lombiq.Privacy

dministro opened the pull request:
OSOE-93

dministro committed:
Updating submodule Lombiq.Privacy, project structure, sample project skeleton
Adding Lombiq.Privacy.Samples to solution

dministro committed:
Updating submodule Lombiq.Privacy, adding sample
Adding Lombiq.Privacy sample entry to Readme.md

dministro committed:
Updating submodules:
 Lombiq.Privacy, UI tests, fix
 Lombiq.UITestingToolbox, extensions
Adding UI test for Lombiq.Privacy

dministro committed:
Merge remote-tracking branch 'remotes/origin/dev' into issue/OSOE-93

dministro committed:
Updating submodules:
 Lombiq.Privacy: complete UI tests, fixes
 Lombiq.UITestingToolbox: fix html-validator config

dministro committed:
Updating submodules:
 Lombiq.ChartJs: stylelint fixes
 Lombiq.Privacy: coding style fixes
 Lombiq.BaseTheme: stylelint fixes

dministro committed:
Updating submodules:
 - Lombiq.ChartJs: updating version
 - Lombiq.Privacy: updating version, add wwwroot
 - Lombiq.BaseTheme: updating version
 - Lombiq.UITestingToolbox: updating version
Adding UI tests and packages to nuget

dministro committed:
Updating submodules:
 - Lombiq.ChartJs: readme
 - Lombiq.Privacy: readmes
Rolling back nuget test projects

sarahelsaig commented:
You have a merge conflict

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-93

sarahelsaig commented on test/Lombiq.OSOCE.Tests.UI/Tests/BehaviorPrivacyTests.cs:
```diff
@@ -0,0 +1,63 @@
+                    await context.TestConsentBannerAsync();
+                },
+                browser)
+            );
```
These two are running in parallel. Is that safe? Also the code is identical besides the theme name, so please extract into a method.

dministro committed:
fixing privacy UI test
updating submodules:
 - Lombiq.ChartJs: fixing License.md
 - Lombiq.Privacy: fixing coding style
 - Lombiq.UITestingToolbox: fixing coding style

dministro committed:
Update submodule:
 - Lombiq.Privacy: adding oc issue link

dministro committed:
Updating submodule:
 - Lombiq.ChartJs: fixing copyright date

dministro committed:
updating submodule:
 - Lombiq.Privacy: comments, readme, navigation

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-93

sarahelsaig reviewed (changes requested):
Add Lombiq.ChartJs.Samples and Lombiq.Privacy.Samples to the [NuGet csproj](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/blob/issue/OSOE-93/NuGetTest/src/Lombiq.OSOCE.NuGet.Web/Lombiq.OSOCE.NuGet.Web.csproj) and invoke the tests for both project from [BehaviorModuleTests.cs](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/blob/issue/OSOE-93/NuGetTest/test/Lombiq.OSOCE.NuGet.Tests.UI/Tests/BehaviorModuleTests.cs) as well. 
(But first push a new tag for Lombiq.Privacy.)

dministro commented:
> Add Lombiq.ChartJs.Samples and Lombiq.Privacy.Samples to the [NuGet csproj](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/blob/issue/OSOE-93/NuGetTest/src/Lombiq.OSOCE.NuGet.Web/Lombiq.OSOCE.NuGet.Web.csproj) and invoke the tests for both project from [BehaviorModuleTests.cs](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/blob/issue/OSOE-93/NuGetTest/test/Lombiq.OSOCE.NuGet.Tests.UI/Tests/BehaviorModuleTests.cs) as well. (But first push a new tag for Lombiq.Privacy.)

I did this [here](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/pull/105/commits/075218320ae89e3fe31c5a4e627a8c7bd5df83e0) but CI `Build and Test (NuGet solution) (Developer)` failed because of the prerelease package versions.

What is the correct process to make this?

sarahelsaig commented:
I have checked out the commit you linked, run the command in the failed CI's log and it worked fine for me with no errors. In principle the errors make no sense, because you have the correct versions in your project files. Could be that you committed too soon and the NuGet cache wasn't updated yet, but normally that results in a "version not found" type of message so I'm just speculating.

Give it another try, but before you push to origin make sure it works for you locally by running
```powershell
dotnet build Lombiq.OSOCE.NuGet.sln --configuration Release --version-suffix 615 /p:Version="1.0.615.0" /p:RunAnalyzersDuringBuild=true /p:TreatWarningsAsErrors=true -warnaserror /nodeReuse:false /p:Retries=4 /p:RetryDelayMilliseconds=1000 /p:NoWarn=AD0001%3BNETSDK1179 -warnAsMessage:MSB3026
```
inside the solution's NuGetTest directory. If it works for you locally and still fails on the server, then open a thread in Teams in the infrastructure channel.

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-93

dministro committed:
Updating submodules:
 - Lombiq.Privacy: merge

dministro committed:
Updating submodules:
 - Lombiq.ChartJs: merge
 - Lombiq.Privacy: merge
 - Lombiq.BaseTheme: merge

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-93

dministro committed:
Updating submodule:
 - Lombiq.ChartJs: rollback

dministro committed:
Updating submodule:
 - Lombiq.Privacy: fixing after merge

dministro committed:
adding UI tests to NuGet
updating submodules:
 - Lombiq.ChartJs: manifest version
 - Lombiq.Privacy: updating Lombiq.Tests.UI nuget version

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-93

dministro committed:
updating submodules:
 - Lombiq.ChartJs: add things to failure dump
 - Lombiq.UITestingToolbox: experimental implementation of adding things to failure dump

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-93

dministro committed:
updating submodule:
 - Lombiq.UITestingToolbox: merge

dministro committed:
updating submodule:
 - Lombiq.UITestingToolbox: disabling hw acceleration in chrome driver

dministro committed:
updating submodules:
 - Lombiq.ChartJs: coding style
 - Lombiq.UITestingToolbox: comments
updating nuget package versions

sarahelsaig commented on NuGetTest/test/Lombiq.OSOCE.NuGet.Tests.UI/.htmlvalidate.json:
```diff
@@ -14,5 +14,16 @@
-  }
+  },
+
+  "elements": [
+    "html5",
+    {
+      "form": {
+        "attributes": {
+          "method": [ "get", "post", "GET", "POST" ]
+        }
+      }
+    }
+  ]
```
Do you still need these? The `dev` of Lombiq.Tests.UI [already uses html-validation v.7.1.1](https://github.com/Lombiq/UI-Testing-Toolbox/blob/beebd4b322cf520c060fe409955b33ac6681a7c8/Lombiq.Tests.UI/Services/HtmlValidationConfiguration.cs#L36).

dministro committed:
updating nuget package versions
removing unnecessary .htmlvalidate.json configuration
updating submodules:
 - Lombiq.ChartJs: nuget package version
 - Lombiq.UITestingToolbox: .htmlvalidate.json, warning suppress annotation

sarahelsaig committed:
update submodules
